docs/.vuepress/tools/watch.mjs

```diff
@@ -0,0 +1,157 @@
+ * Kills the process and whole pid group, then starts a new one
+ * Listens for the success message from vuepress and init or reloads the browser
+ * Browser stack is creating a proxy to the vuepress server.
+ *
+ * @param {ChildProcess} docProcess - Spawn process of the vuepress server.
+ * @returns {ChildProcess}
+ */
+const restartProcess = (docProcess) => {
+  if (docProcess) {
+    process.kill(-docProcess.pid);
+    docProcess = null;
+  }
+  const newDocProcess = spawn(
+    'npm',
+    [
+      'run',
+      process.env.VUEPRESS_NO_CACHE ? 'docs:start:no-cache' : 'docs:start',
+    ],
+    {
+      stdio: 'pipe',
+      detached: true,
+    }
+  );
+
+  newDocProcess.stdin.setEncoding('utf-8');
+  newDocProcess.stdout.pipe(process.stdout);
+
+  if (newDocProcess) {
+    newDocProcess.stdout.on('data', (data) => {
+      // vue press dev is returning patter `success [10:06:15] Build 01050a finished in 83 ms! ( http://localhost:8080/docs/ )`
+      const test = new RegExp('success.*(http://localhost.*/)');
+
+      if (data.toString().match(test)) {
+        const url = data.toString().match(test)[1];
+
+        if (bs.active) {
+          bs.reload();
+        } else {
+          bs.init({
+            proxy: url,
+          });
+        }
+      }
+    });
+  }
+
+  return newDocProcess;
+};
+
+let docProcess = restartProcess();
+
+/** ......................................
+ * Listen for exit events then calls the exitHandler.
+ *
+ * @param {Function} exitHandler - Function to be called on exit.
+ */
+const onExitListener = (exitHandler) => {
+  // do something when app is closing
+  process.on('exit', exitHandler.bind(null, { cleanup: true }));
+  // catches ctrl+c event
+  process.on('SIGINT', exitHandler.bind(null, { exit: true }));
+  // catches "kill pid" (for example: nodemon restart)
+  process.on('SIGUSR1', exitHandler.bind(null, { exit: true }));
+  process.on('SIGUSR2', exitHandler.bind(null, { exit: true }));
+  // catches uncaught exceptions
+  process.on('uncaughtException', exitHandler.bind(null, { exit: true }));
+};
+
+/**
+ * Restarts the server by calling the `restartProcess` function.
+ */
+const restartServer = () => {
+  console.log('Restarting the server');
+  docProcess = restartProcess(docProcess);
+};
+
+/**
+ * Used as method for onExitListener
+ * Gracefully destroys the spawn process and exits the current process.
+ *
+ * @param {object} options - Options for the exitHandler.
+ * @param {string} exitCode - Process exit code.
+ */
+const exitHandler = (options, exitCode) => {
+  if (docProcess) {
+    process.kill(-docProcess.pid);
+    console.log('BYE BYE Handsontable developer !!!', 'Exit code', exitCode);
+    docProcess = null;
+  }
+  if (options.exit) { process.exit(); }
+};
+
+onExitListener(exitHandler);
+
+/**
+ * Watch for changes in the content folder and restart the server in content *.(html|js|css) files.
+ */
+watcherJs.on('change', (path/* , _stat */) => {
+  debounce(restartServer, 1000)();
```

Comment:
Debouncing does not have an effect here. For each file change, there is a new timeout created. So if you applied changes faster than 1000ms, then the server will be restarted twice or more.